rekall-core/rekall/plugins/linux/heap_analysis.py

```diff
@@ -0,0 +1,4388 @@
+
+                            temp_vma['ebp'] = offsets['ebp']
+                            temp_vma['esp'] = offsets['esp']
+
+                            heap_area = False
+
+            temp_vma['name'] = fname
+            temp_vma['vma'] = vma
+            result.append(temp_vma)
+
+        return sorted(result, key=lambda vma: vma['vma'].vm_start)
+
+
+    def _load_libc_profile(self):
+        """Loads the Libc profile for the current libc version. """
+
+        try:
+            if self.__glibc_profile_filename:
+                libc_profile_file = open(self.__glibc_profile_filename, 'r').read()
+                libc_json_profile = json.loads(libc_profile_file)
+                self.session.logging.info("Loading libc profile: " + self.__glibc_profile_filename)
+
+                self._libc_profile = self.profile.LoadProfileFromData(
+                    libc_json_profile,
+                    name='libc',
+                    session=self.session)
+
+        except IOError as ex:
+            self.session.logging.error("Libc profile read error: %s", ex)
+
+        except Exception as ex:
+            self.session.logging.error("Fatal Error: %s", ex)
+            if self.session.GetParameter("debug"):
+                pdb.post_mortem()
+
+        if not self._libc_profile:
+            # we try to gather version information from the libc filename
+            libc_version = None
+            match = None
+            libc_filename = get_libc_filename(self.vmas)
+
+            if libc_filename:
+                match = re.search(r'(\d)\.(\d+)', libc_filename)
+
+            if match and len(match.groups()) == 2:
+                if match.group(1) == '2':
+                    if int(match.group(2)) >= 25:
+                        libc_version = '2.25'
+
+                    elif 23 <= int(match.group(2)) <= 24:
+                        libc_version = '2.23'
+
+                    else:
+                        libc_version = '2.20'
+
+            if not libc_version:
+                # Fallback
+                libc_version = '2.25'
+
+            self.session.logging.info("Loading profile for version {:s}.".format(libc_version))
+
+            if self.session.profile.metadata("arch") == 'I386':
+                self._libc_profile = GlibcProfile32(version=libc_version, session=self.session)
+
+            elif self.session.profile.metadata("arch") == 'AMD64':
+                self._libc_profile = GlibcProfile64(version=libc_version, session=self.session)
+
+
+        if not self._libc_profile:
+            self.session.logging.error('Unable to load a libc profile.')
+
+        else:
+            # merging profiles with self.profile.merge(self._libc_profile) leads to weird side
+            # effects (such as vm_area_struct names getting screwed up). Not sure at the moment
+            # why. Hence, we use those profiles separately, resp. merge them the other way around
+            self._libc_profile.merge(self.profile)
```

Comment:
my thinking here was: i wanted to keep the libc_profile short
to be able to unpack integers and so on without having to include them in libc profiles (also for people using the plugin and providing their own profiles), i merged the two (otherwise i get errors such as "Cant find object unsigned int in profile")
but if there is a better way to do it, i would love to do it that way. do you have a reference in another plugin, where i can get a rough idea?